R/recycle.R

```diff
@@ -1,31 +1,164 @@
-#' Vector recycling
+#' Recycling rules used by r-lib and the tidyverse
+#'
+#' @description
+#' Recycling describes the concept of repeating elements of one vector to match
+#' the size of another vector. There are two rules that guide the recycling
+#' process used by packages in r-lib and the tidyverse:
+#'
+#' - Vectors of size 1 will be recycled to the size of any other vector
+#'
+#' - Otherwise, all vectors must have the same size
+#'
+#' @section Examples:
```

Comment:
```suggestion
#' Recycling describes the concept of repeating elements of one vector to match
#' the size of another. There are two rules that underlying the "tidyverse" recycling
#' rules:
#'
#' - Vectors of size 1 will be recycled to the size of any other vector
#'
#' - Otherwise, all vectors must have the same size
#'
#' @section Examples:
```

Reply:
And then maybe say "These are stricter than the recycling rules generally used by base R, which are ... ", moving the content from below.